scanner/enricher/nvd/nvd.go

```diff
@@ -0,0 +1,421 @@
+	}
+	if cfg.FeedRoot != nil {
+		if !strings.HasSuffix(*cfg.FeedRoot, "/") {
+			return fmt.Errorf("URL missing trailing slash: %q", *cfg.FeedRoot)
+		}
+		u, err := url.Parse(*cfg.FeedRoot)
+		if err != nil {
+			return err
+		}
+		e.feed = u
+	} else {
+		var err error
+		e.feed, err = defaultFeed.Parse(".")
+		utils.CrashOnError(err)
+	}
+	e.feed.RawQuery = "noRejected"
+	zlog.Info(ctx).
+		Str("feed", e.feed.String()).
+		Bool("has_api_key", e.apiKey != "").
+		Dur("call_interval", e.callInterval).
+		Msg("enricher configured")
+	return nil
+}
+
+// Name implements driver.Enricher and driver.EnrichmentUpdater.
+func (*Enricher) Name() string {
+	return name
+}
+
+// FetchEnrichment implements driver.EnrichmentUpdater.
+func (e *Enricher) FetchEnrichment(ctx context.Context, hint driver.Fingerprint) (io.ReadCloser, driver.Fingerprint, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "enricher/nvd/Enricher/FetchEnrichment")
+	out, err := os.CreateTemp("", "enricher.nvd.")
+	if err != nil {
+		return nil, hint, err
+	}
+	utils.Should(os.Remove(out.Name()))
+	var success bool
+	defer func() {
+		if !success {
+			if err := out.Close(); err != nil {
+				zlog.Warn(ctx).Err(err).Msg("unable to close spool")
+			}
+		}
+	}()
+	// Doing this serially is slower, but much less complicated than using an
+	// ErrGroup or the like.
+	//
+	// It may become an issue in 25-30 years.
+	startDate := time.Date(firstYear, time.January, 1, 0, 0, 0, 0, time.UTC)
+	now := time.Now().UTC()
+	for startDate.Before(now) {
+		// The maximum allowable range when using any date range parameters is 120 consecutive days.
+		endDate := startDate.Add(120*24*time.Hour - time.Second)
+		if endDate.After(now) {
+			endDate = now
+		}
+		var apiResp *schema.CVEAPIJSON20
+		for startIdx := 0; ; startIdx += apiResp.TotalResults {
+			apiResp, err = e.query(ctx, startDate, endDate, startIdx)
+			if err != nil {
+				return nil, hint, err
+			}
+			if apiResp.ResultsPerPage == 0 {
+				break
+			}
+			// Parse vulnerabilities in the API response.
+			enc := json.NewEncoder(out)
+			for _, vuln := range apiResp.Vulnerabilities {
+				item := filterFields(vuln.CVE)
+				enrichment, err := json.Marshal(item)
+				if err != nil {
+					return nil, hint, fmt.Errorf("serializing CVE %s: %w", item.ID, err)
+				}
+				r := driver.EnrichmentRecord{
+					Tags:       []string{item.ID},
+					Enrichment: enrichment,
+				}
+				if err := enc.Encode(&r); err != nil {
+					return nil, hint, fmt.Errorf("encoding enrichment: %w", err)
+				}
+			}
+			zlog.Info(ctx).Int("count", len(apiResp.Vulnerabilities)).Msg("loaded vulnerabilities")
+			// Rudimentary rate-limiting.
+			time.Sleep(e.callInterval)
+		}
+		startDate = endDate.Add(time.Second)
+	}
+	// Reset so clients can read the items.
+	if _, err := out.Seek(0, io.SeekStart); err != nil {
+		return nil, hint, fmt.Errorf("unable to reset item feed: %w", err)
+	}
+	success = true
+	// Hint is currently ignored, and always the same.
```

Comment:
what is the implication of keeping this the same?

Reply:
I don´t think we handle the hint in the enricher fetcher.